src/platform/pdf/templates/medications.js

```diff
@@ -0,0 +1,391 @@
+/**
+ * Lab and Test Results PDF template.
+ *
+ * NB: The order in which items are added to the document is important,
+ * and thus PDFKit requires performing operations synchronously.
+ */
+/* eslint-disable no-await-in-loop */
+
+import { MissingFieldsException } from '../utils/exceptions/MissingFieldsException';
+
+import {
+  createAccessibleDoc,
+  createArtifactText,
+  addHorizontalRule,
+  createDetailItem,
+  createHeading,
+  createSpan,
+  createSubHeading,
+  registerVaGovFonts,
+  createImageDetailItem,
+} from './utils';
+
+const config = {
+  margins: {
+    top: 40,
+    bottom: 32,
+    left: 16,
+    right: 32,
+  },
+  headings: {
+    H1: {
+      font: 'Bitter-Bold',
+      size: 24,
+    },
+    H2: {
+      font: 'Bitter-Bold',
+      size: 18,
+    },
+    H3: {
+      font: 'Bitter-Bold',
+      size: 16,
+    },
+    H4: {
+      font: 'Bitter-Bold',
+      size: 14,
+    },
+  },
+  subHeading: {
+    font: 'SourceSansPro-Regular',
+    size: 12,
+  },
+  text: {
+    boldFont: 'SourceSansPro-Bold',
+    font: 'SourceSansPro-Regular',
+    size: 12,
+  },
+};
+
+const generateIntroductionContent = async (doc, parent, data) => {
+  const introduction = doc.struct('Sect', {
+    title: 'Introduction',
+  });
+  parent.add(introduction);
+
+  // title
+  introduction.add(
+    createHeading(doc, 'H1', config, data.title, { x: 16, paragraphGap: 10 }),
+  );
+
+  // preface
+  if (data.preface) {
+    introduction.add(createSubHeading(doc, config, data.preface, { x: 16 }));
+  }
+
+  doc.moveDown();
+  introduction.end();
+};
+
+const generateResultsMedicationListContent = async (
+  medication,
+  doc,
+  results,
+  hasHorizontalRule,
+) => {
+  // medication header
+  if (medication.header) {
+    results.add(
+      await createHeading(doc, 'H3', config, medication.header, {
+        paragraphGap: 10,
+        x: 16,
+      }),
+    );
+  }
+
+  // medication section header
+  for (const section of medication.sections) {
+    results.add(
+      await createHeading(doc, 'H4', config, section.header, {
+        paragraphGap: 10,
+        x: 16,
+      }),
+    );
+
+    // medication section items
+    for (const resultItem of section.items) {
+      let structs;
+      // image item
+      // TODO: check integration when CORS issue is resolved
+      if (resultItem.value && typeof resultItem.value === 'object') {
+        structs = await createImageDetailItem(doc, config, 32, resultItem);
+        // regular item
+      } else {
+        structs = await createDetailItem(doc, config, 32, resultItem);
+      }
+
+      // TODO: refactor to make it work with images when available
```

Comment:
let's add this to the images backlog issue too :-)

Reply:
Will remove TODOs in a separate PR. Thank you.